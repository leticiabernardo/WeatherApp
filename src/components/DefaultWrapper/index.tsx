import { Container } from '@chakra-ui/react';

type WrapperProps = {
  children: React.ReactNode;
  backgroundImage?: string;
};

const defaultProps = {
  backgroundImage: undefined,
};

const DefaultWrapper = ({ children, backgroundImage }: WrapperProps) => {
  return (
    <Container
      maxW="100%"
      width="100%"
      height="100%"
      backgroundImage={backgroundImage ? `url(${backgroundImage})` : undefined}
      backdropFilter="contrast(80%)"
      backgroundSize="cover"
      backgroundPosition="center"
      backgroundAttachment="fixed"
      _after={{
        bgColor: 'rgba(0,0,0,0.3)',
        position: 'fixed',
        zIndex: '2',
        width: '100%',
        height: '100%',
        top: 0,
        left: 0,
        right: 0,
        bottom: 0,
        display: 'block',
        content: '""',
      }}
      display={{ base: 'inline-table', lg: 'block' }}
      padding="0px"
    >
      <Container
        maxW="1100px"
        mx="auto"
        width="100%"
        height={{ base: 'auto', lg: '100%' }}
        zIndex="100"
        position="relative"
      >
        {children}
      </Container>
    </Container>
  );
};

DefaultWrapper.defaultProps = defaultProps;
export default DefaultWrapper;
