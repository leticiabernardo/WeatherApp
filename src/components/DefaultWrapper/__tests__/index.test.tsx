import { render, cleanup } from '@testing-library/react';
import DefaultWrapper from '../index';

const props = {
  children: <div>Test</div>,
};

describe('default wrapper render', () => {
  afterEach(cleanup);

  test('render the component', () => {
    const { container } = render(<DefaultWrapper {...props} />);
    expect(container).toMatchSnapshot();
  });

  test('should contain the text of the children prop', () => {
    const { getByText } = render(<DefaultWrapper {...props} />);
    expect(getByText('Test')).toBeDefined();
    expect(() => getByText('no exist')).toThrow();
  });

  test('should contain the css with background image', () => {
    const backgroundImage = 'image.jpg';

    const { container } = render(
      <DefaultWrapper {...props} backgroundImage={backgroundImage} />
    );
    expect(container).toMatchSnapshot();
  });
});
