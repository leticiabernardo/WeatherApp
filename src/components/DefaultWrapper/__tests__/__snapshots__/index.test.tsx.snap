// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default wrapper render render the component 1`] = `
<div>
  <div
    class="chakra-container css-1wy61kz"
  >
    <div
      class="chakra-container css-1vggzwa"
    >
      <div>
        Test
      </div>
    </div>
  </div>
</div>
`;

exports[`default wrapper render should contain the css with background image 1`] = `
<div>
  <div
    class="chakra-container css-1v6mmd1"
  >
    <div
      class="chakra-container css-1vggzwa"
    >
      <div>
        Test
      </div>
    </div>
  </div>
</div>
`;
